apps/prairielearn/src/lib/code-caller/code-caller-container.ts

```diff
@@ -280,10 +288,10 @@ export class CodeCallerContainer {
     this.stdinStream?.write('\n');
 
     this.state = IN_CALL;
-    this._checkState();
+    this._checkState(undefined);
     this.debug('exit call()');
 
-    return deferred.promise;
+    return deferred.promise as any as { result: any; output: string };
```

Comment:
A safer way to do this is to use generics when constructing the deferred promise:

```ts
const deferred = deferredPromise<{ result: any; output: string }>();
```

The little `{ result: any; output: string }` type is used often enough that you might consider making that it's own type (`CodeCallerResult`?).

Reply:
Sortof fixed it? had to do `deferred.resolve({ result, output: output ?? '' });` a little higher up.